test/new-e2e/containers/docker_vm_test.go

```diff
@@ -0,0 +1,77 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package containers
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/utils/clients"
+	"github.com/DataDog/datadog-agent/test/new-e2e/utils/credentials"
+	"github.com/DataDog/datadog-agent/test/new-e2e/utils/infra"
+
+	"github.com/vboulineau/pulumi-definitions/aws"
+	"github.com/vboulineau/pulumi-definitions/aws/ec2/ec2"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/auto"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"github.com/stretchr/testify/require"
+)
+
+var userData = `#!/bin/bash
+
+set -ex
+
+export DEBIAN_FRONTEND=noninteractive
+
+apt -y update && apt -y install docker.io
+`
+
+func TestAgentDockerVM(t *testing.T) {
+	config := auto.ConfigMap{}
+	env := aws.NewSandboxEnvironment(config)
+	credentialsManager := credentials.NewManager()
+
+	// Retrieving necessary secrets
+	sshKey, err := credentialsManager.GetCredential(credentials.AWSSSMStore, "agent.ci.awssandbox.ssh")
+	require.NoError(t, err)
+
+	apiKey, err := credentialsManager.GetCredential(credentials.AWSSSMStore, "agent.ci.dev.apikey")
+	require.NoError(t, err)
+
+	stack, err := infra.NewStack(context.Background(), "ci", "ci-agent-docker-vm", config, func(ctx *pulumi.Context) error {
```

Comment:
Should this be a context with a timeout (some default value we expect for stacks to be provisioned within). Tests would naturally timeout, just wondering if providing a value here would make that condition easier to troubleshoot in logs.